common/membership/hashring_test.go

```diff
@@ -116,27 +116,85 @@ func TestRefreshUpdatesRingOnlyWhenRingHasChanged(t *testing.T) {
 	pp := NewMockPeerProvider(ctrl)
 
 	pp.EXPECT().Subscribe(gomock.Any(), gomock.Any()).Times(1)
-	pp.EXPECT().GetMembers("test-service").Times(3)
+	pp.EXPECT().GetMembers("test-service").Times(1).Return(randomHostInfo(3), nil)
```

Comment:
hmm.  any idea why this changed?

`Times(1)` seems correct since these start/refresh/refresh should mean one call during start's refresh, and then the other two are "too soon" and are skipped...:
```go
func (r *ring) refresh() error {
	if r.members.refreshed.After(time.Now().Add(-minRefreshInternal)) {
		// refreshed too frequently
		return nil
	}
```
but this was true before as well, and I don't see any time-manipulating in this test that would have made `Times(3)` correct previously.